### PR TITLE
Cleanup++ ... improve/rework the logging ...

### PR DESCRIPTION
**Pull request by alexAubin**
Sooooo of course after starting one PR I could not resist but start refactoring half of the code >_> ...

So here I am, with an attempt to make both the code and the output less chaotic

This is a single huge commit, sorry about that ... but I'll try to summarize the main changes here : 
- It was painful to have to wait for entire app operations (install, remove, upgrade...) to fully complete before having the log being displayed ... so I rework that part to have a proper "stream" with messages arriving in "real-time"
- The logging logic was absolute chaos. I reduced the complexity and introduced proper helpers to handle the messages (also no more 3-different-logs-with-only-partial-operation-and-wtf)
- tabs -> spaces, and other indentation stuff

Hopefully I did not break too many things in the process (I was able to run tests on a few apps, but maybe there are some super-edge cases not properly covered idk)

The commit specific to this PR is : https://github.com/YunoHost/package_check/pull/90/commits/d6345d530ab4591437bda58f4f6cc7a8c4f145ef

Screenshots of the new output : 

hextris

ffsync

### CODE REVIEW

File: sub_scripts/testing_process.sh [hunk 1]
```diff
@@ -342,9 +342,9 @@ CHECK_URL () {
 					fi
 
 					# Print the first 20 lines of the page
-					ECHO_FORMAT "Extract of the page:\n" "white"
+					ECHO_FORMAT "Extract of the page:\n" "white" >> "$test_result"
 					echo -en "\e[37m"	# Write in 'light grey'
-					lynx -dump -force_html "$script_dir/url_output" | head --lines 20 | tee --append "$test_result"
+					lynx -dump -force_html "$script_dir/url_output" | head --lines 20 >> "$test_result"
```
Comment (kay0u) [inline sub_scripts/testing_process.sh]: I'd like to keep it, because in some case there are false negative errors cf. [Installation in a sub path](https://ci-apps-unstable.yunohost.org/ci/job/711)

```
Real url: https://sous.domain.tld/path/
HTTP code: 620
Title of the page: Dotclear - Error
Extract of the page:
Dotclear

Database problem

   Something went wrong while trying to read the database.


Test url: sous.domain.tld/path/
Real url: https://sous.domain.tld/path/
HTTP code: 620
Title of the page: Dotclear - Error
Extract of the page:
Dotclear

Database problem

   Something went wrong while trying to read the database.
```

In a perfect world we want to detect this and remove false negative errors, but we can't handle all specific errors

Comment (alexAubin) [reply sub_scripts/testing_process.sh]: Yes indeed

Later in the code (at least in that PR but not the other I believe :s ) I added : 

```
    if [[ curl_error -ne 0 ]]
    then
        warning "$test_url_details"
        warning "Page title: $page_title"
        warning "Page extract: $page_extract"
    fi
```

with the idea of displaying the info only if it's not what's expected ... though now that you mention it, the current code only assumes it's a curl_error if it's 4xx or 5xx .. maybe it should also consider 6xx codes (c.f your example) (though I don't even understand how that code exists in the first place but anyway)

**Comment by alexAubin** [comment]
Superseded by #92
